src/BlazorWebView/src/SharedSource/BlazorWebViewServiceCollectionExtensions.cs

```diff
@@ -44,17 +44,40 @@ public static IServiceCollection AddMauiBlazorWebView(this IServiceCollection se
 #elif WEBVIEW2_WPF
 			services.TryAddSingleton<WpfBlazorMarkerService>();
 #endif
+
+#if WEBVIEW2_MAUI
+			return new MauiBlazorWebViewBuilder(services);
+#elif WEBVIEW2_WINFORMS
+			return services;
+#elif WEBVIEW2_WPF
 			return services;
+#endif
 		}
 
 		/// <summary>
 		/// Enables Developer tools on the underlying WebView controls.
 		/// </summary>
 		/// <param name="services">The <see cref="IServiceCollection"/>.</param>
 		/// <returns>The <see cref="IServiceCollection"/>.</returns>
+#if WEBVIEW2_WINFORMS
 		public static IServiceCollection AddBlazorWebViewDeveloperTools(this IServiceCollection services)
+#elif WEBVIEW2_WPF
+		public static IServiceCollection AddBlazorWebViewDeveloperTools(this IServiceCollection services)
+#elif WEBVIEW2_MAUI
+		public static IMauiBlazorWebViewBuilder AddBlazorWebViewDeveloperTools(this IServiceCollection services)
```

Comment:
To be honest this is still a bit of a weird API - having `AddBlazorWebViewDeveloperTools` return the same interface type `IMauiBlazorWebViewBuilder` that we put on `AddMauiBlazorWebView` implies you could equally chain subsequent builder logic on either of those calls.

If we were obsessively pattern-focused, we'd have some other `IBlazorWebViewDeveloperToolsBuilder` type get returned from here, in case there were devtools-specific things to chain onto it. However that seems like an absurdly neurotic design.

I'd personally prefer consider returning `IServiceCollection` from this to be less weird. There are dozens of places in ASP.NET Core where we have DI builders returning `IServiceCollection` so it would be tough to argue that's violating our norms. To be clear, I'm still happy with `AddMauiBlazorWebView` returning `IMauiBlazorWebViewBuilder`, as that makes sense.

Special CC to @javiercn @Eilon here for any other opinions since I think you two were speaking up a lot when we originally brought this up in the meeting.

Reply:
Yeah this is what I was failing to convey in my other comment. I just don't know what the "best" pattern here is, but I felt that returning the same builder interface from both, while hanging off IServiceCollection is odd.